R/otu_table.R

```diff
@@ -21,15 +24,24 @@ read_qiime_otu_table <- function(filepath, commented=TRUE, metadata=TRUE) {
   }
   col_names <- strsplit(header_line, "\t")[[1]]
 
-  col_classes <- rep("numeric", times=length(col_names))
-  col_classes[1] <- "character"
-  if (metadata) {
-    col_classes[length(col_classes)] <- "character"
+  if (use.readr & requireNamespace("readr")) {
+    
+    full_otu_table <- as.data.frame(readr::read_tsv(
+      filepath, col_names = col_names, skip = 2))
+
+  } else {
+    
+    col_classes <- rep("numeric", times=length(col_names))
+    col_classes[1] <- "character"
+    if (metadata) {
+      col_classes[length(col_classes)] <- "character"
+    }
+    
+    full_otu_table <- read.table(
+      f, col.names=col_names, colClasses=col_classes, sep="\t", 
+      quote="", as.is=TRUE, header=FALSE)  
   }
 
-  full_otu_table <- read.table(
-    f, col.names=col_names, colClasses=col_classes, sep="\t", 
-    quote="", as.is=TRUE, header=FALSE)
   close(f)
 
   data_cols <- if (metadata) {
```

Comment:
Don't be afraid to clean this up if it helps with `readr` integration.  There are a few tests that you should be able to run with `devtools` for increased confidence.